src/snowflake/cli/api/utils/definition_rendering.py

```diff
@@ -277,7 +277,9 @@ def _add_defaults_to_definition(original_definition: Definition) -> Definition:
     with context({"skip_validation_on_templates": True}):
         # pass a flag to Pydantic to skip validation for templated scalars
         # populate the defaults
-        project_definition = build_project_definition(**original_definition)
+        project_definition = build_project_definition(
+            **copy.deepcopy(original_definition)
```

Comment:
I haven't see Pydantics modify the passed in object. Is the copy necessary here or is build_project_definition modifying it now?

Reply:
It's necessary because applying mixin was changing YamlOverride to dictionary in original definition thus removing the necessary information.